algebras/algebra/src/test/scala/endpoints4s/algebra/EndpointsTestApi.scala

```diff
@@ -218,4 +218,112 @@ trait EndpointsTestApi extends algebra.Endpoints {
         docs.withSummary(docs.summary.map(_ + " (mapped)"))
       }
 
+  val mappedEndpointLeft: Endpoint[(Int, String, Int, String), Either[Unit, (String, String)]] =
+    endpoint(
+      get(
+        path / "mapped-left" /? qs[Int]("x"),
+        headers = requestHeader("If-None-Match")
+      ),
+      ok(
+        emptyResponse,
+        headers = responseHeader("ETag")
+      ),
+      EndpointDocs().withSummary(Some("Initial summary"))
+    )
+      .mapRequest { endpointRequest =>
+        endpointRequest
+          .addQueryString(qs[Int]("y"))
+          .addHeaders(requestHeader("If-Modified-Since"))
+      }
+      .mapResponse { endpointResponse =>
+        response(NotModified, emptyResponse).orElse(
+          endpointResponse
+            .addHeaders(responseHeader("Last-Modified"))
+        )
+      }
+      .mapDocs { docs =>
+        docs.withSummary(docs.summary.map(_ + " (mapped)"))
+      }
+
+  val mappedEndpointRight: Endpoint[(Int, String, Int, String), Either[Unit, (String, String)]] =
+    endpoint(
+      get(
+        path / "mapped-right" /? qs[Int]("x"),
+        headers = requestHeader("If-None-Match")
+      ),
+      ok(
+        emptyResponse,
+        headers = responseHeader("ETag")
+      ),
+      EndpointDocs().withSummary(Some("Initial summary"))
+    )
+      .mapRequest { endpointRequest =>
+        endpointRequest
+          .addQueryString(qs[Int]("y"))
+          .addHeaders(requestHeader("If-Modified-Since"))
+      }
+      .mapResponse { endpointResponse =>
+        response(NotModified, emptyResponse).orElse(
+          endpointResponse
+            .addHeaders(responseHeader("Last-Modified"))
+        )
+      }
+      .mapDocs { docs =>
+        docs.withSummary(docs.summary.map(_ + " (mapped)"))
+      }
+
+  val errorEndpoint = endpoint(
+    get(
+      path / "error" / "user" / segment[String]() / "description" /? (qs[String]("name") & qs[
+        Int
+      ]("age"))
+    ),
+    ok(textResponse)
+  )
+
+  val emptyResponseErrorEndpoint = endpoint(
+    get(
+      path / "error" / "user" / segment[String]() / "description" /? (qs[String]("name") & qs[
+        Int
+      ]("age"))
+    ),
+    ok(emptyResponse)
+  )
```

Comment:
Those two endpoint definitions are the same except for the response entity. I see that the sub server always returns an empty response (unless I did a mistake). I would expect the tests that call `errorEndpoint` to fail. Am I missing something?

Reply:
I am not sure myself what is tested here, assertions are the same for both endpoints. Same situation is with detailed error endpoints. What change would you like to have here?

Reply:
That’s a good point. After looking at the tests I am not sure what they test either… But at least I see that both tests use the same “mock” server, so what you did is correct.